toonz/sources/toonzfarm/tfarmserver/tfarmserver.cpp

```diff
@@ -702,6 +708,14 @@ void FarmServer::queryHwInfo(HwInfo &hwInfo)
 	len = sizeof(physMemSize);
 	sysctl(mib, 2, &physMemSize, &len, NULL, 0);
 
+#elif defined(LINUX)
+	physMemSize = (TINT64)sysconf(_SC_PHYS_PAGES) * (TINT64)sysconf(_SC_PAGE_SIZE);
```

Comment:
You can also query `_SC_AVPHYS_PAGES` for the number of available pages, but like `_SC_PHYS_PAGES` it can be non-standard, so not available on every system (in particular, other non-Linux kernels that will take this path too).

For total virtual memory it’s a lot trickier, the total is dependent on the architecture, with e.g. 4 GiB on most 32-bit OS, 16 EiB theoretically on x86_64, but actually only 256 TiB on current processors.

On Linux you can query these values from `/proc/meminfo`, which can be a bit tricky to interpret but there is some good documentation online about it.  Also remember it’s Linux-only, and won’t be portable to other systems.